Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -74,10 +74,50 @@ public class ResourcesProjectMapper: ProjectMapping {
             sideEffects.append(sideEffect)
         }
 
+        if project.isExternal,
+           target.supportsSources,
+           target.sources.contains(where: { $0.path.extension == "m" || $0.path.extension == "mm" }),
+           !target.resources.filter({ $0.path.extension != "xcprivacy" }).isEmpty
+        {
+            let (headerFilePath, headerData) = synthesizedObjcHeaderFile(bundleName: bundleName, target: target, project: project)
+
+            let headerHash = try headerData.map(contentHasher.hash)
+            let headerFile = SourceFile(path: headerFilePath, contentHash: headerHash)
+            let headerSideEffect = SideEffectDescriptor.file(.init(path: headerFilePath, contents: headerData, state: .present))
+
+            let gccPrefixHeader = "$(SRCROOT)/\(headerFile.path.relative(to: project.path).pathString)"
+            var settings = modifiedTarget.settings?.base ?? SettingsDictionary()
+            settings["GCC_PREFIX_HEADER"] = .string(gccPrefixHeader)
```

Comment:
This solution is not ideal as we're overriding this setting: https://github.com/tuist/tuist/pull/4902#discussion_r1038443248

However, this setting is not really used by SPM targets and we don't apply this for internal targets. If this becomes an issue, we can rethink this. I did try using the `-include` C flag but that caused issues due to import of `Foundation` and I haven't explored this further.